lib/filesystem.nix

```diff
@@ -360,52 +363,30 @@ in
       directory,
       ...
     }:
+    assert pathIsDirectory directory;
```

Comment:
Removing this makes the installer test work, tested locally with `nix-build -A nixosTests.installer.simple`. I believe this might be related to https://github.com/NixOS/nix/issues/11681.